code/game/machinery/doors/brigdoors.dm

```diff
@@ -109,13 +98,75 @@
 
 /obj/machinery/door_timer/Destroy()
 	QDEL_NULL(Radio)
-	targets.Cut()
 	prisoner = null
+	targets.Cut()
 	return ..()
 
-//Main door timer loop, if it's timing and time is >0 reduce time by 1.
-// if it's less than 0, open door, reset timer
-// update the door_timer window and the icon
+/obj/machinery/door_timer/proc/update_record(mob/user)
+	var/datum/data/record/R = find_security_record("name", detaineename)
+	Radio.autosay("Detainee [detaineename] has been incarcerated for [timetoset/10] seconds for the charges of, '[detaineecrimes]'\

```

Comment:
If you want to be really good then this should really be a colon instead of a comma to begin a list.